File: 151051/Krupovich/310 & 320/dc_rest/src/main/java/ru/bsuir/controllers/EditorController.java
package ru.bsuir.controllers;

import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import ru.bsuir.dto.request.EditorRequestTo;
import ru.bsuir.dto.response.EditorResponseTo;
import ru.bsuir.exceptions.IllegalFieldException;
import ru.bsuir.services.EditorService;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/editors")
public class EditorController {

    private final EditorService editorService;

    public EditorController(EditorService editorService) {
        this.editorService = editorService;
    }

    @PostMapping
    public ResponseEntity<EditorResponseTo> create(@RequestBody @Valid EditorRequestTo dto) {

        EditorResponseTo response = editorService.createEditor(dto);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    @GetMapping
    public ResponseEntity<List<EditorResponseTo>> getAll() {
        return new ResponseEntity<>(editorService.getAllEditors(), HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<EditorResponseTo> getById(@PathVariable Long id) {
        return editorService.getEditorById(id)
                .map(editor -> new ResponseEntity<>(editor, HttpStatus.OK))
                .orElseGet(() -> ResponseEntity.notFound().build());
    }

    @PutMapping
    public ResponseEntity<EditorResponseTo> update( @RequestBody @Valid EditorRequestTo dto) {
        if (dto.id() == null) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }
        return editorService.updateEditor(dto.id(), dto)
                .map(updatedEditor -> new ResponseEntity<>(updatedEditor, HttpStatus.OK))
                .orElseGet(() -> ResponseEntity.notFound().build());
    }
    @PutMapping("/{id}")
    public ResponseEntity<EditorResponseTo> updateById(@PathVariable Long id, @RequestBody @Valid EditorRequestTo dto) {
        return editorService.updateEditor(id, dto)
                .map(updatedEditor -> new ResponseEntity<>(updatedEditor, HttpStatus.OK))
                .orElseGet(() -> ResponseEntity.notFound().build());
    }



    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        if (editorService.deleteEditor(id)) {
            return ResponseEntity.noContent().build();
        } else {
            return ResponseEntity.notFound().build();
        }
    }
}

File: 251002/Korti/publisher/src/providers/Editor/Editor.Controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpStatus,
  Param,
  ParseIntPipe,
  Put,
  Post,
  HttpCode,
} from '@nestjs/common';
import { EditorService } from './Editor.Service';
import { EditorResponseTo } from './Dto/EditorResponseTo';
import { EditorRequestTo, UpdateEditorDto } from './Dto/EditorRequestTo';

@Controller('api/v1.0/editors')
export class EditorController {
  constructor(private readonly editorService: EditorService) {}

  @Get()
  @HttpCode(HttpStatus.OK)
  async findAll() {
    const editors = await this.editorService.getAllEditors();
    return editors;
  }

  @Get(':id')
  @HttpCode(HttpStatus.OK)
  async findById(@Param('id', ParseIntPipe) idEditor: number) {
    const editor: EditorResponseTo =
      await this.editorService.findById(idEditor);
    return editor;
  }

  @Post()
  @HttpCode(HttpStatus.CREATED)
  async create(@Body() req: EditorRequestTo) {
    const editor: EditorResponseTo = await this.editorService.createEditor(req);
    return editor;
  }

  @Delete(':id')
  @HttpCode(HttpStatus.NO_CONTENT)
  async delete(@Param('id', ParseIntPipe) idEditor: number): Promise<void> {
    await this.editorService.deleteEditor(idEditor);
  }

  @Put()
  @HttpCode(HttpStatus.OK)
  async update(@Body() body: UpdateEditorDto) {
    const editor = await this.editorService.updateEditor(body);
    return editor;
  }

  @Get('article/:articleId')
  @HttpCode(HttpStatus.OK)
  async getByarticleId(@Param('articleId', ParseIntPipe) articleId: number) {
    const editor = await this.editorService.getEditorByArticleId(articleId);
    return editor;
  }
}
